vector/src/main/java/im/vector/app/features/matrixto/SpaceCardHelper.kt

```diff
@@ -0,0 +1,169 @@
+            }
+        })
+    }
+
+    fun render(spaceChildInfo: SpaceChildInfo?,
+               peopleYouKnow: List<User>,
+               matrixLinkCallback: TimelineEventController.UrlClickCallback?,
+               inCard: FragmentMatrixToRoomSpaceCardBinding) {
+        if (spaceChildInfo == null) {
+            inCard.matrixToCardContentVisibility.isVisible = false
+            inCard.matrixToCardButtonLoading.isVisible = true
+        } else {
+            inCard.matrixToCardContentVisibility.isVisible = true
+            inCard.matrixToCardButtonLoading.isVisible = false
+            avatarRenderer.renderSpace(spaceChildInfo.toMatrixItem(), inCard.matrixToCardAvatar)
+            inCard.matrixToCardNameText.setTextOrHide(spaceChildInfo.name)
+            inCard.matrixToBetaTag.isVisible = true
+            inCard.matrixToCardAliasText.setTextOrHide(spaceChildInfo.canonicalAlias)
+            inCard.matrixToCardDescText.setTextOrHide(spaceChildInfo.topic?.linkify(matrixLinkCallback))
+            if (spaceChildInfo.worldReadable) {
+                inCard.matrixToAccessText.setTextOrHide(stringProvider.getString(R.string.public_space))
+                inCard.matrixToAccessImage.isVisible = true
+                inCard.matrixToAccessImage.setImageResource(R.drawable.ic_public_room)
+            } else {
+                inCard.matrixToAccessText.setTextOrHide(stringProvider.getString(R.string.private_space))
+                inCard.matrixToAccessImage.isVisible = true
+                inCard.matrixToAccessImage.setImageResource(R.drawable.ic_room_private)
+            }
+            val memberCount = spaceChildInfo.activeMemberCount ?: 0
+            if (memberCount != 0) {
+                inCard.matrixToMemberPills.isVisible = true
+                inCard.spaceChildMemberCountText.text = stringProvider.getQuantityString(R.plurals.room_title_members, memberCount, memberCount)
+            } else {
+                // hide the pill
+                inCard.matrixToMemberPills.isVisible = false
+            }
+
+            val images = listOf(
+                    inCard.knownMember1,
+                    inCard.knownMember2,
+                    inCard.knownMember3,
+                    inCard.knownMember4,
+                    inCard.knownMember5
+            ).onEach { it.isGone = true }
+
+            if (peopleYouKnow.isEmpty()) {
+                inCard.peopleYouMayKnowText.isVisible = false
+            } else {
+                peopleYouKnow.forEachIndexed { index, item ->
+                    images[index].isVisible = true
+                    avatarRenderer.render(item.toMatrixItem(), images[index])
+                }
+                inCard.peopleYouMayKnowText.setTextOrHide(
+                        stringProvider.getQuantityString(R.plurals.space_people_you_know,
+                                peopleYouKnow.count(),
+                                peopleYouKnow.count()
+                        )
```

Comment:
There is probably some duplicated code, maybe extract it to private fun.

Reply:
done